src/tests/MDSMatrixAddersTest/MDSMatrixAddersTester.py

```diff
@@ -0,0 +1,165 @@
+        for j in range(12):
+            if j < rand_size:
+                rand_vec.append(random.randint(0, basic.p - 1))
+            else:
+                rand_vec.append(0)
+        rand_matrix.append(rand_vec)
+
+    if rand_size == 5:
+        rand_matrix.append(12 * [0])
+    return rand_valid, rand_size, rand_matrix
+
+
+def set_input_ports(dut, values0, values1, values2):
+    for i in range(3):
+        for j in range(12):
+            exec(
+                "dut.io_inputs_{}_payload_state_elements_{}.value = values{}[{}]".format(
+                    i, j, i, j
+                )
+            )
+
+
+def check_output_ports(dut, ref_res):
+    dut_res = []
+
+    for i in range(12):
+        exec(
+            "dut_res.append(dut.io_output_payload_state_elements_{}.value.integer)".format(
+                i
+            )
+        )
+
+    for i in range(12):
+        if dut_res[i] != ref_res[i]:
+            return False, dut_res
+    return True, dut_res
+
+
+async def generate_input(dut):
+    """generate input signals"""
+    cases_count = 0
+    while cases_count < cases_num:
+        # get random values
+        valid, size, matrix = get_random_input()
+        # assign random values to dut input ports
+        dut.io_inputs_0_valid.value = valid
+        dut.io_inputs_1_valid.value = valid
+        dut.io_inputs_2_valid.value = valid
+        dut.io_inputs_0_payload_state_size.value = size
+        dut.io_inputs_1_payload_state_size.value = size
+        dut.io_inputs_2_payload_state_size.value = size
+
+        if valid == False:
+            await RisingEdge(dut.clk)
+
+        if valid == True:
+            cases_count += 1
+            for i in range(ceil(size / 3)):
+                set_input_ports(
+                    dut, matrix[i * 3 + 0], matrix[i * 3 + 1], matrix[i * 3 + 2]
+                )
+                await RisingEdge(dut.clk)
+                while (
+                    dut.io_inputs_0_valid.value & dut.io_inputs_0_ready.value == False
+                ):
+                    await RisingEdge(dut.clk)
+
+            # calculate the reference outputs
+            ref_res = [0] * 12
+            for i in range(size):
+                for j in range(12):
+                    ref_res[j] = basic.add(ref_res[j], matrix[i][j])
+
+            ref_outputs.put([size, ref_res])
+
+
+async def output_check(dut):
+    """check output signals"""
+    cases_count = 0
+
+    while cases_count < cases_num:
+        await RisingEdge(dut.clk)
+        # get random ready signals
+        ready = random.random() > 0.4
+        dut.io_output_ready.value = ready
+
+        if (dut.io_output_ready.value & dut.io_output_valid.value) == True:
+
+            cases_count += 1
+            size, ref_res = ref_outputs.get()
+
+            passed, dut_res = check_output_ports(dut, ref_res)
+            if not passed:
+                print(
+                    "failed info: case_num:{} input size:{}".format(cases_count, size)
```

Comment:
f"Hello, {name}. You are {age}."

use Python 3 f-string for print format

Reply:
fixed